consensus/fbft_log.go

```diff
@@ -65,127 +61,160 @@ func (m *FBFTMessage) String() string {
 	)
 }
 
+// Hash return the hash of the FBFT message for the unique identifier for FBFT message
+func (m *FBFTMessage) Hash() common.Hash {
+	return hash.FromRLPNew256(m)
+}
+
+// FBFTLog represents the log stored by a node during FBFT process
+type FBFTLog struct {
+	blocks         map[common.Hash]*types.Block // store blocks received in FBFT
+	messages       map[common.Hash]*FBFTMessage // store messages received in FBFT
+	verifiedBlocks map[common.Hash]struct{}     // store block hashes for blocks that has already been verified
+
+	lock sync.Mutex
```

Comment:
how about use separate locks to avoid conflict?

Reply:
Do you mean one lock for each of the map? I thought this is not the performance bottle neck so I didn't optimize this. Do you think this might be the potential bottleneck?

Reply:
Different sync mutex on different fields is done. Please check the latest code.